push/src/curl_wrapper.cc

```diff
@@ -22,12 +22,18 @@ CurlWrapper::CurlWrapper(const std::string& username,
     throw std::runtime_error("Cannot initialize easy curl!");
   }
 
+  optHttpHeader_ = nullptr;
+
   if (!username.empty()) {
     auth_ = username + ":" + password;
   }
 }
 
 CurlWrapper::~CurlWrapper() {
+  if (optHttpHeader_ != nullptr)
```

Comment:
No need to check for null. The `curl_slist_free_all` function is a no-op if the pointer is null